### ISSUE TITLE
[xcm] Investigate the possibility of `ProcessXcmMessage` / `XcmExecutor` emitting an event on failure.

### ISSUE THREAD

**Issue by bkontur** [paritytech/polkadot-sdk#2780]
RIght now, we can see just `success = false` for `MesssageQueue` event in case of failure.

Relates to: https://github.com/paritytech/polkadot-sdk/issues/478

**Comment by ggwpez** [comment]
IIRC the original idea was that the message processor can emit events to notify the outside world of why an error happened.  
Currently we use a [freestanding struct](https://github.com/paritytech/polkadot-sdk/blob/e1c033ebe15e6de5ee6a5a81eab2dd213f09d574/polkadot/xcm/xcm-builder/src/process_xcm_message.rs#L47) as processor, hence it cannot emit events.  
Now there could be a pallet (or using pallet-xcm) that emits an event in the error case. Not sure if this is ideal, but it is easier than trying to have an associated and aggregated error type.  

Maybe instead of returning a `bool` it could also return an `impl ProcessingError` that can be converted to a `bool` but also has an `ToString` which can then be included in the error event from the MQ pallet. Sounds a bit hacky though since it would use strings.

**Comment by xlc** [comment]
who is this for? devs or indexers or users?
if you just want to be able to debug issues, just add bunch logs

**Comment by bkchr** [comment]
Yeah it was about devs and logs are being added here: https://github.com/paritytech/polkadot-sdk/pull/2799

**Comment by acatangiu** [comment]
@xlc @bkchr logs only help in replays (try-runtime, chopsticks), but don't really improve UX for live chains - where users/dapps can see "live" why something fails.

If adding events is unwanted for some reason, maybe we can look at improving/enhancing error types and error messages (right now a lot of failure reasons result in same generic error).

Let me know what you think so we either close it or start working on smth here.

**Comment by xlc** [comment]
we could have new runtime API to dry run xcm and return detailed errors

**Comment by raymondkfcheung** [comment]
`XcmEventEmitter::emit_send_failure_event()` on PR #7234 partially addresses this issue.  It tries to emit a `SendFailed` event, but due to rollback behaviour, the event is not persisted—only a log message is recorded.  

PR #7730 introduces nested errors in `pallet-xcm`, which is a step toward more detailed failure reporting. Further improvements, such as structured errors or runtime APIs for dry-running XCM, could also help fully resolve the issue.

**Comment by acatangiu** [comment]
This issue was tracking adding detailed error event for "live" failures on-chain. And #7234 does that. We can close this issue 👍
